File: Assignment1/app/src/main/java/com/example/assignment1/MainActivity.java
package com.example.assignment1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private Button signButton;
    private EditText usernameText, emailText, passwordText, confirmText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        signButton = findViewById(R.id.signButton);
        usernameText = findViewById(R.id.usernameText);
        emailText = findViewById(R.id.emailText);
        passwordText = findViewById(R.id.passwordText);
        confirmText = findViewById(R.id.confirmText);





        signButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String username1 = usernameText.getText().toString();
                String email1 = emailText.getText().toString();
                String password1 = passwordText.getText().toString();
                String confirmPass1 = confirmText.getText().toString();
                Intent intent = new Intent(MainActivity.this,SecondActivity.class);
                intent.putExtra("username",username1);
                intent.putExtra("email",email1);
                intent.putExtra("password",password1);
                intent.putExtra("confirmPassword",confirmPass1);
                startActivity(intent);

            }
        });


    }
}

File: CustomCallingScreen/app/src/main/java/com/example/customcallingscreen/activity/MainActivity.kt
package com.example.customcallingscreen.activity

import android.Manifest
import android.content.Intent
import android.os.Bundle
import android.telecom.TelecomManager
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.PermissionChecker
import androidx.core.net.toUri
import com.example.customcallingscreen.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private val REQUEST_PERMISSION: Int = 0;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        callButton.setOnClickListener(View.OnClickListener {
            makeCall()
        })
    }


    override fun onStart() {
        super.onStart()
        offerReplacingDefaultDialer()
    }

    private fun makeCall() {
        if (PermissionChecker.checkSelfPermission(
                this,
                Manifest.permission.CALL_PHONE
            ) == PermissionChecker.PERMISSION_GRANTED
        ) {
            val uri = "tel:${phoneNoEditText.text.toString()}".toUri()
            startActivity(Intent(Intent.ACTION_CALL, uri))
        } else {
            ActivityCompat.requestPermissions(
                this,
                arrayOf(Manifest.permission.CALL_PHONE),
                REQUEST_PERMISSION
            )
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        if (requestCode == REQUEST_PERMISSION && PermissionChecker.PERMISSION_GRANTED in grantResults) {
            makeCall()
        }
    }

    private fun offerReplacingDefaultDialer() {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            getSystemService(TelecomManager::class.java).defaultDialerPackage != packageName
            Intent(TelecomManager.ACTION_CHANGE_DEFAULT_DIALER)
                .putExtra(TelecomManager.EXTRA_CHANGE_DEFAULT_DIALER_PACKAGE_NAME, packageName)
                .let(::startActivity)
        } else {
            Toast.makeText(
                this,
                "Your device does not support this functionality",
                Toast.LENGTH_SHORT
            ).show()
        }
    }
}